pom.xml

```
@@ -52,11 +52,13 @@
 
 	<properties>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-		<maven.compiler.source>8</maven.compiler.source>
-		<maven.compiler.target>8</maven.compiler.target>
+		<maven.compiler.source>17</maven.compiler.source>
+		<maven.compiler.target>17</maven.compiler.target>
 
 		<!-- Dependencies -->
-		<spring-boot.version>2.7.10</spring-boot.version>
+		<spring-boot.version>3.0.5</spring-boot.version>
+		<jakarta.annotation-api.version>2.1.1</jakarta.annotation-api.version>
```

We may omit the version, defined in the Spring Boot BoM.

```suggestion
		<spring-boot.version>3.0.5</spring-boot.version>
```

---

Good idea.